src/main/java/com/tozny/sdk/UserApi.java

```diff
@@ -0,0 +1,280 @@
+     */
+    public Result linkResult(String otp) {
+        Map<String, String> args = new HashMap<String, String>() {{
+            put("otp", otp);
+        }};
+
+        return this.<Result>rawCall("user.link_result", args, new TypeReference<Result>() {});
+    }
+
+    /**
+     * Validate an email or SMS-based OTP in userland.
+     *
+     * @param otp        The OTP provided to the end user
+     * @param session_id The session ID from the original *.otp_challenge request.
+     *
+     * @return A Realm-signed version of the OTP session.
+     */
+    public Result otpResult(String otp, String session_id) {
+        Map<String, String> args = new HashMap<String, String>() {{
+            put("otp", otp);
+            put("session_id", session_id);
+        }};
+
+        return this.<Result>rawCall("user.otp_result", args, new TypeReference<Result>() {});
+    }
+
+    /**
+     * Exchange a signed magic link enrollment challenge for an actual enrollment challenge.
+     *
+     * @param signed_data Signed magic link payload
+     * @param signature   Realm-signed signature
+     *
+     * @return An object representing a device enrollment challenge
+     */
+    public EnrollmentChallenge enrollmentChallengeExchange(String signed_data, String signature) {
+        Map<String, String> args = new HashMap<String, String>() {{
+            put("signed_data", signed_data);
+            put("signature", signature);
+        }};
+
+        return this.<EnrollmentChallenge>rawCall("user.challenge_exchange", args, new TypeReference<EnrollmentChallenge>() {});
+    }
+
+    /**
+     * Exchange a signed magic link authentication challenge for a signed user information pauload
+     *
+     * @param signed_data Signed magic link payload
+     * @param signature   Realm-signed signature
+     *
+     * @return An object representing the user that has just authenticated
+     */
+    public AuthenticationChallenge authenticationChallengeExchange(String signed_data, String signature, String session_id) {
+        Map<String, String> args = new HashMap<String, String>() {{
+            put("signed_data", signed_data);
+            put("signature", signature);
+            put("session_id", session_id);
+        }};
+
+        return this.<AuthenticationChallenge>rawCall("user.challenge_exchange", args, new TypeReference<AuthenticationChallenge>() {});
+    }
+
+    /**
+     * Send a raw method call to the Tozny API
+     *
+     * @param method       Name of the method to invoke
+     * @param parameters   Map of the additional parameters to send with the invocationd
+     * @param valueTypeRef The object type to use when deserializing the response
+     * @param <T>          Descendant of ToznyAPIResponse
+     *
+     * @return Instance of the specified data class.
+     *
+     * @throws ToznyApiException if an I/O or protocol error occurs
+     */
+    private <T extends ToznyApiResponse<?>> T rawCall(
```

Comment:
Stole it from the `dispatch()` method in `RealmAPI`. Wish I could take credit ... but nah.